alpakatest and alpaka fail to build

From https://github.com/cms-patatrack/pixeltrack-standalone/pull/120#issuecomment-724907150

> The `alpaka` and `alpakatest` fail to build with
> ```
> .../pixeltrack-standalone/external/alpaka/include/alpaka/core/Concepts.hpp:81:33: error: '__T0' was not declared in this scope
>                  static_assert(std::is_base_of<type, TDerived>::value, "The type implementing the concept has to be a publicly accessible base class!");
>                                  ^~~~
> .../pixeltrack-standalone/external/alpaka/include/alpaka/core/Concepts.hpp:81:33: note: suggested alternative: '__y0'
>                  static_assert(std::is_base_of<type, TDerived>::value, "The type implementing the concept has to be a publicly accessible base class!");
>                                  ^~~~
>                                  __y0
> .../pixeltrack-standalone/external/alpaka/include/alpaka/core/Concepts.hpp:81:47: error: template argument 1 is invalid
>                  static_assert(std::is_base_of<type, TDerived>::value, "The type implementing the concept has to be a publicly accessible base class!");
> ```
> 
> Updating Alpaka to 0.5.0 changed the error to
> ```
> .../pixeltrack-standalone/external/alpaka/include/alpaka/core/UniformCudaHip.hpp: In instantiation of 'void alpaka::uniform_cuda_hip::detail::rtCheckIgnore(const Error_t&, const char*, const char*, const int&, TErrors&& ...) [with TErrors = {cudaError}; alpaka
> ::uniform_cuda_hip::detail::Error_t = cudaError]':
> .../pixeltrack-standalone/external/alpaka/include/alpaka/queue/cuda_hip/QueueUniformCudaHipRtBase.hpp:196:342:   required from here
> .../pixeltrack-standalone/external/alpaka/include/alpaka/core/UniformCudaHip.hpp:86:61: error: missing braces around initializer for 'std::__array_traits<cudaError, 1>::_Type' {aka 'cudaError [1]'} [-Werror=missing-braces]
>                      std::array<Error_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{ignoredErrorCodes...};
>                                                              ^~~~~~~~~~~~~~~~~~
> ```
> which can be worked around.

--- a/Makefile
+++ b/Makefile
@@ -95,7 +95,8 @@ export BOOST_LDFLAGS := -L$(BOOST_BASE)/lib
 ALPAKA_BASE := $(EXTERNAL_BASE)/alpaka
 export ALPAKA_DEPS := $(ALPAKA_BASE)
 export ALPAKA_CXXFLAGS := -I$(ALPAKA_BASE)/include
-export ALPAKA_CUFLAGS := $(CUDA_CUFLAGS)
+# Temporarily filter out missing-braces warning, see https://github.com/cms-patatrack/pixeltrack-standalone/issues/126
+export ALPAKA_CUFLAGS := $(filter-out -Werror=missing-braces,$(CUDA_CUFLAGS))
 
 CUPLA_BASE := $(EXTERNAL_BASE)/cupla
 export CUPLA_DEPS := $(CUPLA_BASE)/lib
@@ -194,8 +195,6 @@ endif
 
 # Targets and their dependencies on externals
 TARGETS_ALL := $(notdir $(wildcard $(SRC_DIR)/*))
-# Temporarily filter out programs that do not build (yet) with CUDA 11
-TARGETS_ALL := $(filter-out alpakatest alpaka,$(TARGETS_ALL))
 # Temporarily filter out programs that do not run (yet) with CUDA 11
 TARGETS_ALL := $(filter-out cuda cudauvm,$(TARGETS_ALL))
 
@@ -382,7 +381,7 @@ $(BOOST_BASE):
 external_alpaka: $(ALPAKA_BASE)
 
 $(ALPAKA_BASE):
-	git clone git@github.com:alpaka-group/alpaka.git -b release-0.4.1 $@
+	git clone git@github.com:alpaka-group/alpaka.git -b 0.5.0 $@
 
 # Cupla
 .PHONY: external_cupla

